src/components/molecules/UserForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+import { useAppDispatch } from '../../app/hooks'
+import { addUserAction } from '../../features/users/action'
+
+interface IFormInput {
+  firstName: string
+  lastName: string
+  email: string
+  password: string
+  address: string
+}
+
+const schema = yup.object().shape({
+  firstName: yup
+    .string()
+    .min(2, 'First name too short')
+    .required('First name is required'),
+  lastName: yup
+    .string()
+    .min(2, 'First name too short')
+    .required('First name is required'),
+  email: yup.string().email('Invild email').required('Email is required'),
+  password: yup.string().required('Password is required'),
+  address: yup.string().required('Password is required'),
+})
+
+export const UserForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<IFormInput>({ resolver: yupResolver(schema) })
+
+  const dispatch = useAppDispatch()
+
+  const onSubmit = (data: IFormInput) => {
+    if (data !== undefined) {
```

Comment:
this check is optional, data couldnt be underfined, because you provide validation